www/js/services/qrcode_parser.js

```diff
@@ -0,0 +1,54 @@
+(function(window) {
+
+  var COMPANY_INDICATOR = '-';
+  var PERSONAL_INDICATOR = '.';
+
+  var QRCodeParser = function(url) {
+  };
+
+  QRCodeParser.prototype.setUrl = function(url) {
+    this.url = new URL (url);
+  };
```

Comment:
This class would be much simpler I think if you just use a regular expression to match against the URL.